packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -239,11 +239,35 @@ export class InternalModelManager {
   constructor(
     private readonly apiIdentifier: string,
     readonly connection: GadgetConnection,
-    readonly options?: { pluralApiIdentifier: string }
+    readonly options?: { pluralApiIdentifier: string; hasAmbiguousIdentifiers: boolean }
   ) {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
+  private validateRecord(record: RecordData) {
+    if (!this.options || !this.options.hasAmbiguousIdentifiers) {
+      return true;
+    }
+    const keys = Object.keys(record);
+
+    // any keys that aren't the apiIdentifier are invalid
+    return !keys.some((key) => key !== this.apiIdentifier);
+  }
+
+  private getRecordFromData(record: RecordData, functionName: string) {
+    let recordData = record;
+
+    if (!this.validateRecord(record)) {
+      throw Error(`Invalid arguments found in variables. Did you mean to use ${functionName}({ ${this.apiIdentifier}: { ... } })?`);
+    }
+
+    if (this.apiIdentifier in record) {
+      recordData = recordData[this.apiIdentifier];
```

Comment:
👍 looks like we don't have any codepath that could have extra params in the internal API right now, but we'll have to remember that if that changes in the future this would potentially break (since it would clobber those params). Very likely we'd catch that kind of change in CI.